iroh-net/src/util.rs

```diff
@@ -29,3 +29,43 @@ impl<T> Drop for AbortingJoinHandle<T> {
         self.0.abort();
     }
 }
+
+impl<T> std::ops::Deref for AbortingJoinHandle<T> {
+    type Target = tokio::task::JoinHandle<T>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// Holds a handle to abort a task.
+///
+/// The task can be manually aborted, or aborted once self is dropped.
+/// See [`tokio::task::AbortHandle`].
+pub struct CancelOnDrop {
```

Comment:
good call to start creating a util for this, we have this in several places by now.